node/src/sub_lib/stream_key.rs

```diff
@@ -106,21 +111,87 @@ impl StreamKey {
 
 type HashType = [u8; sha1::DIGEST_LENGTH];
 
+fn add_socket_addr_to_hash(mut hash: sha1::Sha1, socket_addr: SocketAddr) -> sha1::Sha1 {
+    match socket_addr {
+        SocketAddr::V4(v4) => {
+            hash.update(v4.ip().octets().as_slice());
+        }
+        SocketAddr::V6(v6) => {
+            hash.update(v6.ip().octets().as_slice());
```

Comment:
Technically, I don't think the Node officially supports IPv6 yet; but when it does, this will be one less thing to change.